core-web/libs/edit-content/src/lib/feature/edit-content/store/features/content.feature.ts

```diff
@@ -0,0 +1,253 @@
+import { tapResponse } from '@ngrx/operators';
+import {
+    patchState,
+    signalStoreFeature,
+    type,
+    withComputed,
+    withMethods,
+    withState
+} from '@ngrx/signals';
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { forkJoin, of, pipe } from 'rxjs';
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { computed, inject } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { switchMap } from 'rxjs/operators';
+
+import {
+    DotContentTypeService,
+    DotHttpErrorManagerService,
+    DotRenderMode,
+    DotWorkflowsActionsService
+} from '@dotcms/data-access';
+import {
+    ComponentStatus,
+    DotCMSContentlet,
+    DotCMSContentType,
+    DotCMSWorkflow,
+    DotCMSWorkflowAction,
+    FeaturedFlags,
+    WorkflowStep
+} from '@dotcms/dotcms-models';
+
+import { WorkflowState } from './workflow.feature';
+
+import { DotEditContentService } from '../../../../services/dot-edit-content.service';
+import { parseWorkflows, transformFormDataFn } from '../../../../utils/functions.util';
+import { EditContentRootState } from '../edit-content.store';
+
+export interface ContentState {
+    /** ContentType full data */
+    contentType: DotCMSContentType | null;
+    /** Contentlet full data */
+    contentlet: DotCMSContentlet | null;
+    /** Schemas available for the content type */
+    schemes: {
+        [key: string]: {
+            scheme: DotCMSWorkflow;
+            actions: DotCMSWorkflowAction[];
+            firstStep: WorkflowStep;
+        };
+    };
+}
+
+export const contentInitialState: ContentState = {
+    contentType: null,
+    contentlet: null,
+    schemes: {}
+};
+
+export function withContent() {
+    return signalStoreFeature(
+        { state: type<EditContentRootState & WorkflowState>() },
+        withState(contentInitialState),
+        withComputed((store) => ({
+            /**
+             * Computed property that determines if the content is new.
+             * Content is considered new when there is no contentlet in the store.
+             *
+             * @returns {boolean} True if content is new, false otherwise
+             */
+            isNew: computed(() => !store.contentlet()),
+
+            /**
+             * Computed property that determines if the store's status is equal to ComponentStatus.LOADED.
+             *
+             * @returns {boolean} - Returns true if the store's status is LOADED, otherwise false.
+             */
+            isLoaded: computed(() => store.state() === ComponentStatus.LOADED),
+
+            /**
+             * A computed property that checks if an error exists in the store.
+             *
+             * @returns {boolean} True if there is an error in the store, false otherwise.
+             */
+            hasError: computed(() => !!store.error()),
```

Comment:
You need to get a way of this kind of boolean state properties because they increase the complexity by a lot https://kentcdodds.com/blog/stop-using-isloading-booleans

Reply:
We have `isLoading`, `isLoaded` and `isSaving` suuuuuper red flag.